### ISSUE TITLE
Specify and implement exception handling strategy for swig wrappers

### ISSUE THREAD

**Issue by prock-fife** [fifengine/fifengine#186]
This issue was reported by **jasoka**

This ticket should decide how exceptions should be handled in fife swig wrappers and implement the decided strategy. Information about exception handling and swig is available in http://www.swig.org/Doc1.3/SWIGPlus.html#SWIGPlus_exception_specifications. If exception handling is not done, swig won't create exception handlers that convert exceptions from c++ to scripting side. Because of that, nice, clean exceptions starts to look like program crash

Basically the task could be to dig out what kind of exceptions might occur. One possibility is naturally mark just very generic exceptions, but I'm not sure if that's a good idea. Still one possibility is to create default exception handler with swig syntax using scripting language variation (since the handler is language specific). However I'm not sure if that causes perf penalty. So there are few options, and this task could also think what is the best way to do it.
